### ISSUE TITLE
Move `pyransac3d` to optional dependency

### ISSUE THREAD

**Issue by rhugonnet** [GlacioHack/xdem#726]
It's a somewhat small package used only for a specific functionality used in a single xDEM function. So it would fit well as optional, along `noisyopt` or `opencv` (the latter to be removed, because too big a dependency).

Incidentally, `pyransac3d` does not exist on `conda`, we won't be able to release there with it as a main dependency anyway.

**Comment by rhugonnet** [comment]
Alternatively, if the objective is to fit an outlier-resilient 2/3D plane, we might have existing functionalities for this in `biascorr.py` and `fit.py` we could re-use, already used for Deramping: https://xdem.readthedocs.io/en/stable/biascorr.html#deramping?

@adebardo Do you know if we could re-use `sklearn.linear.RANSACRegressor` that's already used as optional import (here: https://github.com/GlacioHack/xdem/blob/main/xdem/fit.py#L40), or if this does not work in 3D? 🤔 If we can minimize the number of optional dependencies, might as well.
Sometimes we also simply run the optimization with an outlier-robust loss function: https://github.com/GlacioHack/xdem/blob/main/xdem/fit.py#L72.

**Comment by rhugonnet** [comment]
Because we can now pass any `fit_optimizer` manually for all `Coreg` classes, we should probably remove the `sklearn` optional dependency and simply exemplify in the doc how to pass any custom optimizer.

It would be a good to have a good all-rounder optimizer that is outlier-robust we can set as default (to mirror the NMAD, Dowd's variogram, used as default too).

**Comment by adebardo** [comment]
Hi @rhugonnet ,
After confirmation, we can indeed recommend the use of `sklearn.linear_model.RANSACRegressor.`
I’ve opened a PR for that.

In the meantime, while we're waiting for the complete removal of sklearn, I think we can open a ticket for drafting a new all-rounder optimizer. This will allow us to track the proposals and refinements more precisely than in this issue, which mainly deals with the movement of pyransac.

**Comment by rhugonnet** [comment]
Resolved by #727
